haskell-cabal.el

```diff
@@ -1052,6 +1052,62 @@ source-section."
     ;; unwind
     (haskell-mode-toggle-interactive-prompt-state t)))
 
-(provide 'haskell-cabal)
 
+(defun haskell-cabal--find-tags-dir ()
+  "Return a directory where TAGS file will be generated.
+Tries to find cabal file first and if succeeds uses its location.
+If cabal file not found uses current file directory.  If current
+buffer not visiting a file returns nil."
+  (or (haskell-cabal-find-dir)
+      (when buffer-file-name
+        (file-name-directory buffer-file-name))))
+
+(defun haskell-cabal--compose-hasktags-command (&optional cabal-dir)
```

Comment:
Please make `cabal-dir` non-optional, always require this to be a proper directory.